File: leetcode/remove-duplicates-from-sorted-array-ii.py
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 0;

        int last = 0;
        int previous = last;

        for (int i = 0; i < nums.length; i++) {
            if (i == 0) {
                last = nums[i];
                k++;
            } else {
                if (i == 1 || last != nums[i] || previous != last) {
                    nums[k] = nums[i];
                    k++;
                }

                previous = last;
                last = nums[i];
            } 
        } 

        return k;
    }
}

File: leetcode/remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count = 1;
        int index = 1;

        for (size_t i = 1; i < nums.size(); ++i) {
            if (nums.at(i) == nums.at(i - 1)) {
                ++count;
            } else {
                count = 1;
            }

            if (count <= 2) {
                nums.at(index) = nums.at(i);
                ++index;
            }
        }

        return index;
    }
};